desktop/renderer/index.js

```diff
@@ -94,19 +96,20 @@ ReactDOM.render(
   appEl,
 )
 
-if (module.hot) {
-  module.hot.accept('./container', () => {
+module.hot && module.hot.accept('./container', () => {
+  try {
+    store.dispatch({type: updateReloading, payload: {reloading: true}})
     const NewRoot = require('./container').default
     ReactDOM.render(
       <AppContainer><NewRoot store={store} /></AppContainer>,
       appEl,
     )
-  })
-}
+    engine.reset()
+  } finally {
+    setTimeout(() => store.dispatch({type: updateReloading, payload: {reloading: false}}), 10e3)
```

Comment:
Without this timeout reloading is set to false before any actions really run, so it doesn't do anything.

This is a bit hacky, but this is also dev-only. And it's not bad if actions think they aren't reloading, just inefficient.